assignment6/src/ParkingTest.java

```diff
@@ -0,0 +1,59 @@
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+
+public class ParkingTest {
+    private Parking parking = new Parking();
+
+    @Test
+    public void testNoElements() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{0}, new int[]{0});
+        assertEquals(0, movements.size());
+    }
+
+    @Test
+    public void testOneElementSamePositions() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{0, 1}, new int[]{0, 1});
+        assertEquals(0, movements.size());
+    }
+
+    @Test
+    public void testOneElementDifferentPositions() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{1, 0}, new int[]{0, 1});
+        checkEntries(new int[][]{{0, 1}}, movements);
+    }
+
+    @Test
+    public void testTwoElementsWithTwoMovements() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{2, 1, 0}, new int[]{0, 2, 1});
+        checkEntries(new int[][]{{1, 2}, {0, 1}}, movements);
+    }
+
+    @Test
+    public void testMoreElementsWithoutSamePositionOfZero() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{1, 2, 0, 3}, new int[]{3, 1, 2, 0});
+        checkEntries(new int[][]{{1, 2}, {0, 1}, {3, 0}}, movements);
+    }
+
+    @Test
+    public void testMoreElementsWithSamePositionOfZero() {
+        List<Movement> movements = parking.RearrangingCars(new int[]{1, 2, 0, 3}, new int[]{3, 1, 0, 2});
+        checkEntries(new int[][]{{0, 2}, {3, 0}, {1, 3}, {2, 1}}, movements);
+    }
+
+    private void checkEntries(int[][] realMovements, List<Movement> expectedMovements) {
+        int size = expectedMovements.size();
+        assertThat(size, is(realMovements.length));
+        for (int i = 0; i < size; i++) {
+            Entry<Integer, Integer> entry = expectedMovements.get(i).getValues();
+            assertThat(new int[]{entry.getKey(), entry.getValue()}, is(realMovements[i]));
+        }
+    }
+}
```

Comment:
You have some good tests on small input, and well-named tests (it's useful to know you're testing the case where zero moves and doesn't move - these are good things to cover), but I would have liked to see some more tests - maybe larger input, or testing a larger array moved cyclically one place left and right, or a long array reversed, or swapped elementwise. Think of things that should stress the algorithm and find corner-case problems, and test them.